### ISSUE TITLE
Cannot login to Keycloak with HTTPS from Windows (but can from Linux)

### ISSUE THREAD

**Issue by user1537** [keycloak/keycloak#11010]
### Describe the bug

We have Keycloak 17.0.1 (Quarkus) running primarily in Kubernetes, but we also tried it on a virtual machine. Both use Java 11 to run Keycloak and are configured to use HTTPS with a self-signed certificate (generated using keytool). In Kubernetes we tried both our custom image based on the newest UBI8, and also the official image. The virtual machine runs RHEL 7.

The issue is that we are unable to log in to administration console from Windows machines (tried from three different computers) - when trying to log in, the process always ends with this message (both with correct and incorrect credentials):

![Unexpected-error](https://user-images.githubusercontent.com/102288468/160843506-1fbb24a3-bad8-4cb9-a23b-c117e02c5002.png)

We are able to log in to the same Keycloak instance from Linux (Fedora 35) systems though (also tried three different systems, all of which worked).

When the Keycloak server runs with HTTP (http-enabled:true; hostname-strict-https:false), we can successfully log in to the administration console from all the Windows and Linux systems.

We tried both Chrome and Firefox browsers, and the behaviour is the same.

### Version

17.0.1 (Quarkus)

### Expected behavior

Being able to log in to administration console normally.

### Actual behavior

When trying to log in from the Windows machines, the server returns status code 400 - this is the request URL:

```
https://<ip>:<port>/realms/master/login-actions/authenticate?session_code=fUdxAl_f-vYHQFGUCX07SQnWEnncBCPf8hi3t3pmmHs&execution=05d68330-cbf2-4b28-9d04-2365989934d1&client_id=security-admin-console&tab_id=eK7ABwblVp8
```

In Keycloak logs, this exception is shown (truncated where the three dots are for brevity):

```
2022-03-23 14:45:41,653 WARN  [org.keycloak.services] (executor-thread-6) KC-SERVICES0013: Failed authentication: java.lang.RuntimeException: java.io.IOException: Underlying input stream returned zero bytes
    at org.jboss.resteasy.plugins.server.BaseHttpRequest.getFormParameters(BaseHttpRequest.java:61)
    at org.jboss.resteasy.plugins.server.BaseHttpRequest.getDecodedFormParameters(BaseHttpRequest.java:74)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
    at org.jboss.resteasy.core.ContextParameterInjector$GenericDelegatingProxy.invoke(ContextParameterInjector.java:166)
    at com.sun.proxy.$Proxy46.getDecodedFormParameters(Unknown Source)
    at org.keycloak.authentication.DefaultAuthenticationFlow.processAction(DefaultAuthenticationFlow.java:104)
    at org.keycloak.authentication.AuthenticationProcessor.authenticationAction(AuthenticationProcessor.java:1002)
    at org.keycloak.services.resources.LoginActionsService.processFlow(LoginActionsService.java:321)
    at org.keycloak.services.resources.LoginActionsService.processAuthentication(LoginActionsService.java:292)
    at org.keycloak.services.resources.LoginActionsService.authenticate(LoginActionsService.java:276)
    at org.keycloak.services.resources.LoginActionsService.authenticateForm(LoginActionsService.java:349)
...
Caused by: java.io.IOException: Underlying input stream returned zero bytes
    at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:288)
    at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
    at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
    at java.base/java.io.InputStreamReader.read(InputStreamReader.java:181)
    at java.base/java.io.BufferedReader.fill(BufferedReader.java:161)
    at java.base/java.io.BufferedReader.read1(BufferedReader.java:212)
    at java.base/java.io.BufferedReader.read(BufferedReader.java:287)
    at org.jboss.resteasy.plugins.providers.FormUrlEncodedProvider.parseForm(FormUrlEncodedProvider.java:88)
    at org.jboss.resteasy.plugins.server.BaseHttpRequest.getFormParameters(BaseHttpRequest.java:57)
```

### How to Reproduce?

We deployed Keycloak to the above mentioned RHEL 7 virtual machine in this way:

We unpacked the 17.0.1 zip file. Then we generated a new pkcs12 certificate using keytool, and started the server using this script:

```
#!/bin/bash

export KEYCLOAK_ADMIN=keycloak
export KEYCLOAK_ADMIN_PASSWORD=keycloak

/root/keycloak-17.0.1/bin/kc.sh start \
--https-key-store-file="/opt/keycloak/conf/keystore.p12" \
--https-key-store-password="pass123" \
--hostname="10.230.10.67" \
--hostname-port="8443"
```

No other changes / configuration was done. Immediately after deploying the server, we tried to log in to the administration console from a Windows machine, which produced the issue described in this ticket.

### Anything else?

I compared the two outgoing authentication requests (from Windows and from Fedora) in Chrome's Network tab, and they looked almost identical: small difference in accepted-language header - but only the lowest priority one, (expected) differences in session cookies.

I already created a discussion [here](https://github.com/keycloak/keycloak/discussions/10931), because I didn't necessarily think this was a bug, but I am increasingly convinced that it just might be. And it seems that this happens to another user as well (according to their comment on the discussion thread linked above) - but only on SOME Windows machines.

Chrome version used:
Version 99.0.4844.84 (Official Build) (64-bit)

Windows version used:
Edition: Windows 10 Enterprise
Version: 21H2

**Comment by Jess103** [comment]
I get "Underlying input stream returned zero bytes" error for HTTPS too.
I tried 17, 17.0.1, and 18 and all not working.
I set proxy=edge, still not working.
port=443, and using root ca certificate.

https://keycloak-url/admin/console
shows the page all well
but when I type ID/PW and click Login button, it throws me that error.

I'm using JDK, quarkus (kc.sh) one for the keycloak server (CentOS7)
And I use Windows PC to get into admin console.
http://keycloak-url/admin/console works perfectly fine.
Why only https not working????
